hera_mc/cm_redis_corr.py

```diff
@@ -0,0 +1,160 @@
+    of correlator mappings
+
+    Parameters
+    ----------
+    cminfo : dict
+        Dictionary as returned from get_cminfo_correlator()
+    redis_info : None or dict
+        Pass through of redis information to retrieve snap hostname
+
+    Returns
+    -------
+    snap_to_ant : dict
+        Dictionary mapping the snaps to the antennas
+    ant_to_snap : dict
+        Dictionary mapping each antenna to its snap
+    """
+    snap_to_ant = {}
+    ant_to_snap = {}
+    for antn, ant in enumerate(cminfo['antenna_numbers']):
+        name = cminfo['antenna_names'][antn]
+        pol_info = {}
+        for pol_snap in cminfo['correlator_inputs'][antn]:
+            pol = pol_snap.lower()[0]
+            if pol in ['e', 'n'] and '>' in pol_snap:
+                pol_info[pol] = pol_snap
+        ant_to_snap[ant] = {}
+        for pol, psnap in six.iteritems(pol_info):
+            snapi, channel = snap_part_to_host_input(psnap, redis_info=redis_info)
+            ant_to_snap[ant][pol] = {'host': snapi, 'channel': channel}
+            snap_to_ant.setdefault(snapi, [None] * 6)
+            snap_to_ant[snapi][channel] = name + pol.upper()
+    return snap_to_ant, ant_to_snap
+
+
+def cminfo_redis_loc(cminfo):
+    """
+    Places the positional data into redis for use by the correlator.
+
+    Parameter
+    ---------
+    cminfo : dict
+        Dictionary as returned from get_cminfo_correlator()
+
+    Returns
+    -------
+    ant_pos : dict
+        Dictionary containing the ECEF positions of all of the antennas.
+    cofa : dict
+        Dictionary containing the lat, lon, alt of the center-of-array
+    """
+    locations = {}
+    locations['antenna_numbers'] = cminfo['antenna_numbers']
+    locations['antenna_names'] = cminfo['antenna_names']
+    antenna_positions = []
+    for _x in cminfo['antenna_positions']:
+        antenna_positions.append(list(_x))
+    locations['antenna_positions'] = antenna_positions
+    locations['antenna_utm_eastings'] = cminfo['antenna_utm_eastings']
+    locations['antenna_utm_northings'] = cminfo['antenna_utm_northings']
+    locations['cofa'] = {'lat': cminfo['cofa_lat'], 'lon': cminfo['cofa_lon'],
+                         'alt': cminfo['cofa_alt']}
+    return locations
+
+
+def set_redis_cminfo(redishost=DEFAULT_REDIS_ADDRESS, session=None, testing=False):
+    """
+    Gets the configuration management information and writes to the
+    redis database for the correlator.
+
+    Parameters
+    ----------
+    redishost : None or str
+        Hostname for the redis database.  If None uses default
+    session : None or hera_mc session
+        Session for hera_mc instance.  None uses default
+    """
+    h = cm_sysutils.Handling(session=session)
+    cminfo = h.get_cminfo_correlator()
+
+    # This is retained so that explicitly providing redishost=None has the desired behavior
+    if redishost is None:
+        redishost = DEFAULT_REDIS_ADDRESS
```

Comment:
This line isn't covered either, but maybe we can live with it not covered. Or add a pragma.